v-next/hardhat-utils/src/eth.ts

```diff
@@ -1,19 +1,61 @@
 import type { PrefixedHexString } from "./hex.js";
 
-import { bytesToHexString, numberToHexString, setLengthLeft } from "./hex.js";
+import { LinkReferenceError } from "./errors/eth.js";
+import {
+  bytesToHexString,
+  getPrefixedHexString,
+  getUnprefixedHexString,
+  numberToHexString,
+  setLengthLeft,
+} from "./hex.js";
 import {
   getAddressGenerator,
   getHashGenerator,
   isValidChecksum,
 } from "./internal/eth.js";
 
+/**
+ * Represents an Ethereum address.
+ *
+ * Note: TypeScript does not have a way to represent strings with a fixed
+ * length, so we use an alias to represent addresses starting with "0x".
+ * If you need to validate the address, you can use the `isAddress` function.
+ */
+export type Address = PrefixedHexString;
+
+/**
+ * Represents a hash value.
+ *
+ * Note: TypeScript does not have a way to represent strings with a fixed
+ * length, so we use an alias to represent hashes starting with "0x".
+ * If you need to validate the hash, you can use the `isHash` function.
+ */
+export type Hash = PrefixedHexString;
+
+/**
+ * A map of library names to their addresses. The keys can be contract names or
+ * fully qualified names in the form `sourceName:contractName`.
+ */
+export interface LibraryAddresses {
```

Comment:
Where do we use this?

Reply:
This is currently used in `hardhat-viem`: the user can pass a map used for linking the contract's bytecode. I put it here because I believe we have something similar elsewhere (either `hardhat-ethers` or `hardhat-verify`). But, as this is part of the public api, I think it's actually a mistake and the type should live in the plugin's code. I'll change it.